charts/prometheus-alerts/templates/containers-prometheusrule.yaml

```diff
@@ -480,4 +480,154 @@ spec:
         {{- end }}
     {{- end }}
 
+  - name: {{ .Release.Name }}.{{ .Release.Namespace }}.alertRulesSelectorsValidity
```

Comment:
Unless I am misreading - the problem I see here is that this chart is launched in tons of places and it creates rules that may never be used .. ie, it creates DaemonSet rules even if the namespace has no DaemonSets. That's not _great_ .. but it previously didn't cause any problems. Now we're talking about creating an alarm that I think will fire off for almost everyone when it gets created because they are mostly not using all of these resources.

Eg.. I just ran this with some fake selectors:

<img width="1710" alt="image" src="https://github.com/Nextdoor/k8s-charts/assets/768067/9d3eb188-7f78-4716-8984-89abfe68ec30">

For the `simple-app`, `daemonset-app`, `rollout-app` charts, we could go down the route of adding in options that simply turn off the unrelated alarms (ie... add in `.Values.containerRules.enableDaemonsetRules` for example).

For _outside_ callers of this chart, we'll need a big noisy readme breaking change changelog that tells users to add these variables in or they'll get paged.

Thoughts?